SConstruct

```diff
@@ -340,21 +340,6 @@ if env["platform"] not in platform_list:
 
     Exit(0 if env["platform"] == "list" else 255)
 
-
-# FIXME: Tool assignment happening at this stage is a direct consequence of getting the platform logic AFTER the SCons
-# environment was already been constructed. Fixing this would require a broader refactor where all options are setup
-# ahead of time with native validator/converter functions.
-custom_tools = []
-if env["platform"] == "android":
-    custom_tools += ["clang", "clang++", "as", "ar", "link"]
-elif env["platform"] == "web":
-    custom_tools += ["cc", "c++", "ar", "link", "textfile", "zip"]
-elif env["platform"] == "windows" and methods.get_cmdline_bool("use_mingw", False):
-    custom_tools += ["mingw"]
-for tool in custom_tools:
-    env.Tool(tool)
```

Comment:
I moved this further down because there's more options overrides below, notably platform flags, and we likely want to respect them before calling `detect.get_tools(env)` so that this can make full use of the configuration so far.